client/java/src/main/java/org/jolokia/client/request/ValidatingResponseExtractor.java

```diff
@@ -54,10 +54,14 @@ public ValidatingResponseExtractor(int... pCodesAllowed) {
     public <RESP extends J4pResponse<REQ>, REQ extends J4pRequest> RESP extract(REQ pRequest,
                                                                       JSONObject pJsonResp)
             throws J4pRemoteException {
-
-        int status = pJsonResp.containsKey("status") ?
-                ((Long) pJsonResp.get("status")).intValue() :
-                0;
+	
+	int status = 0;
+	if( pJsonResp.containsKey("status")) {
+		Object o = pJsonResp.get("status");
+		if( o instanceof Long ) {
+			status = ((Long)o).intValue();
+		}
```

Comment:
Yes, you are right, but I think its better to include the extra context information which type and (string) value status currently have, like ("Invalid status of type java.lang.String ('error') received"). The more context info in an error, the better.